src/util/configurations.rs

```diff
@@ -0,0 +1,83 @@
+/// This struct should contain all user customizable options as this is passed to the channel to be accessed
+#[derive(Clone, Debug)]
+pub struct UserConfig{
+	/// optional user specified channel limits
+	/// These are only used on startup of channels, and are referenced to a single instance
+	pub channel_limits : ChannelHandshakeLimits,
+	/// channel options are unique to each channel
+	pub channel_options : ChannelConfig,
+}
+
+impl UserConfig {
+	///default constructor, calls ChannelOptions and ChannelLimits constructors
+    pub fn new() -> Self{
+        UserConfig {
+            channel_limits : ChannelHandshakeLimits::new(),
+			channel_options : ChannelConfig::new(),
+        }
+    }
+}
+
+/// This struct contains all the optional bolt 2 channel limits.
+/// If the user wants to check a value, the value needs to be filled in, as by default they are not checked
+#[derive(Copy, Clone, Debug)]
+pub struct ChannelHandshakeLimits{
+	/// minimum allowed funding_satoshis
+	pub min_funding_satoshis :u64,
+	/// maximum allowed htlc_minimum_msat
+	pub max_htlc_minimum_msat : u64,
+	/// min allowed max_htlc_value_in_flight_msat
+	pub min_max_htlc_value_in_flight_msat : u64,
+	/// max allowed channel_reserve_satashis
+	pub max_channel_reserve_satoshis : u64,
+	/// min allowed max_accepted_htlcs
+	pub min_max_accepted_htlcs : u16,
+	/// min allowed dust_limit_satashis
+	pub min_dust_limit_satoshis : u64,
+	///minimum depth to a number of blocks that is considered reasonable to avoid double-spending of the funding transaction
+	pub minimum_depth : u32,
+	///do we force the incoming channel to match our announced channel preference
+	pub force_announced_channel_preference : bool,
+}
+
+impl ChannelHandshakeLimits {
+//creating max and min possible values because if they are not set, means we should not check them.
+///default constructor creates limits so that they are not tested for
+	pub fn new() -> Self{
+		ChannelHandshakeLimits {
+			min_funding_satoshis : 0,
+			max_htlc_minimum_msat : <u64>::max_value(),
+			min_max_htlc_value_in_flight_msat : 0,
+			max_channel_reserve_satoshis : <u64>::max_value(),
+			min_max_accepted_htlcs : 0,
+			min_dust_limit_satoshis : 0,
+			minimum_depth : <u32>::max_value(),
+			force_announced_channel_preference : false,
+		}
+	}
+}
+
+/// This struct contains all the custom channel options.
+#[derive(Copy, Clone, Debug)]
+pub struct ChannelConfig{
+	/// Amount (in millionths of a satoshi) channel will charge per transferred satoshi.
+	pub fee_proportional_millionths : u32,
+	//TODO enforce non-mutability when config can change at runtime
+	///Is this channel an announced channel, cannot change after channel creation
+	pub announced_channel : bool,
+}
+impl ChannelConfig {
+	/// creating a struct with values.
+	/// fee_proportional_millionths should be changed afterwords
+	pub fn new() -> Self{
+		ChannelConfig {
+			fee_proportional_millionths : 0,
+			announced_channel : true,
+		}
+	}
+}
```

Comment:
nit: Dunno what editor you're using, but its missing end-of-file newlines.

Reply:
I am using vscode with the rls plugin.